java/src/hu/holyoil/controller/SunController.java

```diff
@@ -64,16 +65,18 @@ public void StartSunstorm()  {
         Queue<Asteroid> traversingQueue = new LinkedList<>();
 
         Random rand = new Random();
-        traversingQueue.add(
-                asteroids.get(
-                        rand.nextInt(asteroids.size())
-                )
-        );
+        Asteroid startingAsteroid = asteroids.get(rand.nextInt(asteroids.size()));
+        traversingQueue.add(startingAsteroid);
+        chosenAsteroids.add(startingAsteroid);
 
         while (chosenAsteroids.size() < asteroids.size() / 3 && !traversingQueue.isEmpty()) {
            Asteroid currentAsteroid = traversingQueue.remove();
-           chosenAsteroids.addAll(currentAsteroid.GetNeighbours());
-           traversingQueue.addAll(currentAsteroid.GetNeighbours());
+           List<Asteroid> untraversedNeighbours = currentAsteroid.GetNeighbours()
+                   .stream()
+                   .filter(chosenAsteroids::contains)
```

Comment:
filter keeps all elements that the predicate returns true for. This should be negated, because we want to add ones we have not chosen yet

Reply:
lol yes 🍏